Build failure: `hyprsysteminfo`

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

1. Update flake
2. Rebuild

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/310586346

### Relevant log output

```console
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/mdr1dhp84mixpj2488cgdvr5z7yljr7m-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/xj70mqq0v438p9ii443gd4s6hpj3c9bk-hyprsysteminfo-0.1.3/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/xj70mqq0v438p9ii443gd4s6hpj3c9bk-hyprsysteminfo-0.1.3/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/xj70mqq0v438p9ii443gd4s6hpj3c9bk-hyprsysteminfo-0.1.3/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/xj70mqq0v438p9ii443gd4s6hpj3c9bk-hyprsysteminfo-0.1.3/share/doc/hsi -DCMAKE_INSTALL_INFODIR=/nix/store/xj70mqq0v438p9ii443gd4s6hpj3c9bk-hyprsysteminfo-0.1.3/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/xj70mqq0v438p9ii443gd4s6hpj3c9bk-hyprsysteminfo-0.1.3/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/xj70mqq0v438p9ii443gd4s6hpj3c9bk-hyprsysteminfo-0.1.3/include -DCMAKE_INSTALL_SBINDIR=/nix/store/xj70mqq0v438p9ii443gd4s6hpj3c9bk-hyprsysteminfo-0.1.3/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/xj70mqq0v438p9ii443gd4s6hpj3c9bk-hyprsysteminfo-0.1.3/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/xj70mqq0v438p9ii443gd4s6hpj3c9bk-hyprsysteminfo-0.1.3/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/qdwim37sf31raprhmrjgrqfhppl7d5vb-gcc-wrapper-15.2.0/bin/strip -DCMAKE_RANLIB=/nix/store/qdwim37sf31raprhmrjgrqfhppl7d5vb-gcc-wrapper-15.2.0/bin/ranlib -DCMAKE_AR=/nix/store/qdwim37sf31raprhmrjgrqfhppl7d5vb-gcc-wrapper-15.2.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/xj70mqq0v438p9ii443gd4s6hpj3c9bk-hyprsysteminfo-0.1.3
-- The CXX compiler identification is GNU 15.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/qdwim37sf31raprhmrjgrqfhppl7d5vb-gcc-wrapper-15.2.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found OpenGL: /nix/store/as2rg79vk4z7frb7hy27nmadc3fp1a4j-libglvnd-1.7.0/lib/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Found WrapVulkanHeaders: /nix/store/icyb0ffm7i4sajzphnf818firkqlx71h-vulkan-headers-1.4.328.0/include
-- Found Wayland_Client: /nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-client.so (found version "1.24.0")
-- Found Wayland_Server: /nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-server.so (found version "1.24.0")
-- Found Wayland_Cursor: /nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-cursor.so (found version "1.24.0")
-- Found Wayland_Egl: /nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-egl.so (found version "18.1.0")
-- Found Wayland: /nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-client.so;/nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-server.so;/nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-cursor.so;/nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-egl.so (found version "1.24.0")
-- Found WaylandScanner: /nix/store/pd0fbk9yrygb8c5p36j4cazpf6lw8v75-wayland-scanner-1.24.0-bin/bin/wayland-scanner
-- Found PkgConfig: /nix/store/0skggkamysbh0ksc3582lb69wzrpabcg-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
-- Checking for module 'hyprutils'
--   Found hyprutils, version 0.10.0
-- Configuring done (4.2s)
CMake Error at src/CMakeLists.txt:15 (target_link_libraries):
  Target "hyprsysteminfo" links to:

    Qt6::WaylandClientPrivate

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done (0.1s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_C_COMPILER
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.1-zen1, NixOS, 25.11 (Xantusia), 25.11.20251015.544961d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Determinate Nix 3.11.3) 2.31.2`
 - nixpkgs: `/nix/store/pv11fms46b44vv50nvi3l1ckjdlrm6lf-source`

### Notify maintainers

@donovanglover @fufexan @johnrtitor @khaneliman @NotAShelf

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/hy/hyprsysteminfo/package.nix
+++ b/pkgs/by-name/hy/hyprsysteminfo/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch,
   cmake,
   qt6,
   pkg-config,
@@ -22,6 +23,15 @@ stdenv.mkDerivation (finalAttrs: {
     tag = "v${finalAttrs.version}";
     hash = "sha256-KDxT9B+1SATWiZdUBAQvZu17vk3xmyXcw2Zy56bdWbY=";
   };
+
+  patches = [
+    # Fix Qt6::WaylandClientPrivate not found
+    # https://github.com/hyprwm/hyprsysteminfo/pull/21
+    (fetchpatch {
+      url = "https://github.com/hyprwm/hyprsysteminfo/commit/fe81610278676d26ff47f62770ac238220285d3a.patch";
+      hash = "sha256-rfKyV0gkfXEhTcPHlAB+yxZ+92umBV22YOK9aLMMBhM=";
+    })
+  ];
 
   nativeBuildInputs = [
     cmake

